lib/rex/socket.rb

```diff
@@ -222,7 +229,7 @@ def self.gethostbyname(host)
       host, _ = host.split('%', 2)
     end
 
-    ::Socket.gethostbyname(host)
+    @@resolver ? self.rex_gethostbyname(host) : ::Socket.gethostbyname(host)
```

Comment:
If either of these calls fail, we should probably be handling this appropriately vs blindly assigning to @@resolver, particularly given both of these will require reaching out to remote hosts to perform actions.

Reply:
What do you mean? Only one of those calls (`self.rex_gethostbyname(host)`) can occur when `@@resolver` is defined, otherwise it will use the underlying `Socket` method

Reply:
I honestly don't know why I was thinking assignment was involved here. I blame looking through this when I was too tired 😓 Sorry for the confusion, marking this as resolved.